java/vector/src/main/codegen/templates/FixedValueVectors.java

```diff
@@ -482,7 +482,15 @@ public long getTwoAsLong(int index) {
 
     </#if>
 
-    <#if minor.class == "Date">
+    <#if minor.class == "DateDay">
+    @Override
+    public ${friendlyType} getObject(int index) {
+        org.joda.time.DateTime date = new org.joda.time.DateTime(0, org.joda.time.DateTimeZone.getDefault())
```

Comment:
Hmm..I am actually not sure if we need to do:

```
org.joda.time.DateTime date = new org.joda.time.DateTime(0,org.joda.time.DateTimeZone.UTC).plusDays(get(index));
date = date.withZoneRetainFields(org.joda.time.DateTimeZone.getDefault());
```
plusDays might change hour for daylight saving?
http://joda-time.sourceforge.net/apidocs/org/joda/time/DateTime.html#plusDays(int)